include/picongpu/fields/incidentField/profiles/DispersivePulse.hpp

```diff
@@ -299,15 +299,39 @@ namespace picongpu
                             float_X const dt = static_cast<float_X>(picongpu::SI::DELTA_T_SI / UNIT_TIME);
                             // interpolation order
                             float_X N_raw = Unitless::INIT_TIME / dt;
-                            int n = static_cast<int>(N_raw * 0.5_X); // -0 instead of -1 for rounding up N_raw
+                            int const n = static_cast<int>(N_raw * 0.5_X); // -0 instead of -1 for rounding up N_raw
+
+                            // frequency step for DFT
+                            float_X const dOmk = pmacc::math::Pi<float_X>::doubleValue / Unitless::INIT_TIME;
+
+                            // Since the (Gaussian) spectrum has only significant values near the central frequency,
+                            // the summation over all frequencies is reduced to frequencies within an interval
+                            // around the central frequency.
+
+                            // standard deviation of the Gaussian distributed spectrum
+                            float_X const sigma_Om = 1._X / (pmacc::math::sqrt(2._X) * Unitless::PULSE_DURATION);
+
+                            // index of the mean frequency of the Gaussian distributed spectrum
+                            // unit: [dOmk]
+                            int const center_k
+                                = static_cast<int>(SPEED_OF_LIGHT * Unitless::INIT_TIME / Unitless::Base::WAVE_LENGTH);
+
+                            // index of the lowest frequency in the Gaussian distributed spectrum which is used in the
+                            // DFT 4*sigma_Om distance from central frequency
+                            int const minOm_k = center_k - static_cast<int>(4._X * sigma_Om / dOmk);
+                            int const k_min = minOm_k < 1 ? 1 : minOm_k;
+
+                            // index of the highest frequency in the Gaussian distributed spectrum which is used in the
+                            // DFT
+                            int const maxOm_k = 2 * center_k - minOm_k;
+                            int const k_max = maxOm_k <= n ? maxOm_k : n;
 
                             float_X E_t = 0.0_X; // electric field in time-domain
-                            float_X Omk = 0.0_X; // stores angular frequency for DFT-loop; Omk= 2pi*k/T
+                            float_X Omk = 0.0_X; // stores angular frequency for DFT-loop; Omk = 2pi*k/T = dOmk * k
```

Comment:
please move the definition into the loop because the value is not required outside.